src/main/java/com/exasol/dbbuilder/dialects/AbstractUser.java

```diff
@@ -47,12 +47,12 @@ public String getPassword() {
 
     @Override
     public Map<DatabaseObject, ObjectPrivilege[]> getObjectPrivileges() {
-        return this.objectPrivileges;
+        return Collections.unmodifiableMap(this.objectPrivileges);
```

Comment:
Could we also use `Map.of()`?

Reply:
No, it only allows creating maps from keys & values.
Plus this makes the intention of an unmodifiable map explicit.